website/events/services.py

```diff
@@ -336,3 +337,16 @@ def generate_category_statistics():
         }
 
     return data
+
+
+def execute_data_minimization(dry_run=False):
+    """Delete information about very old events."""
+    # Sometimes years are 366 days of course, but better delete 1 or 2 days early than late
+    deletion_period = timezone.now().date() - timezone.timedelta(days=(365 * 7))
+
+    queryset = EventRegistration.objects.filter(event__end__lte=deletion_period).filter(
+        Q(payment__isnull=False) | Q(member__isnull=False) | ~Q(name__exact="<removed>")
```

Comment:
Does this work for both free events and paid events?

Reply:
Also, does this work for non-member registrations?

Reply:
Finally, event registration information fields should probably be deleted too (though one could argue about this)

Reply:
Read the queryset job :p 

Free and paid events are not different, this just gets _all_ registrations from events that are more than 7 years ago.
First filter gets all those.

Second filter gets all the registrations that match one of the following condititions:
1. The payment is not null
2. The member is not null
3. The name (thus non-member) is not `<removed>`

Agree on removing information fields data.

Maybe that this information can be minimised much earlier than 7 years? We don't need to keep it, the payment information is in the payments themselves and the amount of people that went to the event is preserved using this method.

Reply:
yea i think something like 5 years could be good. it is sometimes nice to look back at old events and it's weird to see "<deleted>" everywhere, i think it would inhibit the nostalgia somewhat. so 5 years could strike the balance

Reply:
Wild idea (maybe too complex though): add a "remove data after" field to registration info fields. As the kind of data in there can differ, maybe our retention policy should also be set per case. For example, when visiting an in-house day at the ministry of defense where we ask for a pasport number, we might want to set a deletion date of the day after the event, while for other information we might want a longer period. This way, we also enforce people to think about when to delete the data (and we could have that field be max. 5 years)

Reply:
Kind of makes sense, but not something that should be done in this PR.

Reply:
Agreed